src/main/java/com/adyen/model/recurring/StoreTokenRequest.java

```diff
@@ -0,0 +1,257 @@
+	 *
+	 * @return recurring
+	 */
+	public Recurring getRecurring() {
+		return recurring;
+	}
+
+	public void setRecurring(Recurring recurring) {
+		this.recurring = recurring;
+	}
+
+	public StoreTokenRequest recurring(Recurring recurring) {
+		this.recurring = recurring;
+		return this;
+	}
+
+	/**
+	 * Some payment methods require defining a value for this field to specify how to process the transaction.
+	 *
+	 * @return selectedBrand
+	 */
+	public String getSelectedBrand() {
+		return selectedBrand;
+	}
+
+	public void setSelectedBrand(String selectedBrand) {
+		this.selectedBrand = selectedBrand;
+	}
+
+	public StoreTokenRequest selectedBrand(String selectedBrand) {
+		this.selectedBrand = selectedBrand;
+		return this;
+	}
+
+	/**
+	 * The shopper's email address.
+	 *
+	 * @return shopperEmail
+	 */
+	public String getShopperEmail() {
+		return shopperEmail;
+	}
+
+	public void setShopperEmail(String shopperEmail) {
+		this.shopperEmail = shopperEmail;
+	}
+
+	public StoreTokenRequest shopperEmail(String shopperEmail) {
+		this.shopperEmail = shopperEmail;
+		return this;
+	}
+
+	/**
+	 * The shopper's IP address.
+	 *
+	 * @return shopperIP
+	 */
+	public String getShopperIP() {
+		return shopperIP;
+	}
+
+	public void setShopperIP(String shopperIP) {
+		this.shopperIP = shopperIP;
+	}
+
+	public StoreTokenRequest shopperIP(String shopperIP) {
+		this.shopperIP = shopperIP;
+		return this;
+	}
+
+	/**
+	 * The shopper's reference for the payment transaction.
+	 *
+	 * @return shopperReference
+	 */
+	public String getShopperReference() {
+		return shopperReference;
+	}
+
+	public void setShopperReference(String shopperReference) {
+		this.shopperReference = shopperReference;
+	}
+
+	public StoreTokenRequest shopperReference(String shopperReference) {
+		this.shopperReference = shopperReference;
+		return this;
+	}
+
+	/**
+	 * The text to appear on a shopper's statement.
+	 *
+	 * @return shopperStatement
+	 */
+	public String getShopperStatement() {
+		return shopperStatement;
+	}
+
+	public void setShopperStatement(String shopperStatement) {
+		this.shopperStatement = shopperStatement;
+	}
+
+	public StoreTokenRequest shopperStatement(String shopperStatement) {
+		this.shopperStatement = shopperStatement;
+		return this;
+	}
+
+	public StoreTokenRequest setCardData(String cardNumber, String cardHolder, String expiryMonth, String expiryYear, String cvc) {
+		Card card = new Card();
+		card.setExpiryMonth(expiryMonth);
+		card.setExpiryYear(expiryYear);
+		card.setHolderName(cardHolder);
+		card.setNumber(cardNumber);
+		card.setCvc(cvc);
+
+		this.setCard(card);
+		return this;
+	}
+
+	public StoreTokenRequest selectRecurringContract() {
+		Recurring recurring = new Recurring();
+		recurring.contract(Recurring.ContractEnum.RECURRING);
+		setRecurring(recurring);
+
+		return this;
+	}
+
+	public StoreTokenRequest selectOneClickContract() {
+		Recurring recurring = new Recurring();
+		recurring.contract(Recurring.ContractEnum.ONECLICK);
```

Comment:
minor : we can replace this by :  
 setRecurring(new Recurring().contract(Recurring.ContractEnum.ONECLICK));